### PR TITLE
Use deterministic OCR configuration

### PR DESCRIPTION
**Pull request by connorwstein**
- Add support for shared secrets in setting OCR config. Use new libocr deterministic function and add OCR secrets as a piece of env specific deployment config. 
- Backwards compat with existing CCIPConfig contract
- Add NonceAt for clearing pending txes CLI

### CODE REVIEW

File: integration-tests/deployment/helpers.go [hunk 1]
```diff
@@ -13,9 +13,26 @@ import (
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/pkg/errors"
 )
 
+type OCRSecrets struct {
```
Comment (krehermann) [inline integration-tests/deployment/helpers.go:24]: code comment please about what and why.

i thought that research was going to provide a func in libocr for this, so i'm confused how who is expected to use it and under what conditions

**Review by krehermann — CHANGES_REQUESTED** [review]
add clarity about use please

File: integration-tests/deployment/ccip/deploy_home_chain.go [hunk 2]
```diff
@@ -316,7 +317,9 @@ func BuildAddDONArgs(
 		if err2 != nil {
 			return nil, err2
 		}
-		signers, transmitters, configF, _, offchainConfigVersion, offchainConfig, err2 := ocr3confighelper.ContractSetConfigArgsForTests(
+		signers, transmitters, configF, _, offchainConfigVersion, offchainConfig, err2 := ocr3confighelper.ContractSetConfigArgsDeterministic(
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/deploy_home_chain.go:321]: @krehermann this is the new libocr function. Its on us to supply common secrets shared across signers - one ephemeral and one not, will have more details on how to construct them for production soon.

File: integration-tests/deployment/helpers.go (hunk 1 above)
Comment (connorwstein) [reply integration-tests/deployment/helpers.go:24]: added a comment for the struct. See below for the new libocr fn

**Review by krehermann — DISMISSED** [review]

**Review by AnieeG — DISMISSED** [review]

**Review by AnieeG — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

**Review by kalverra — APPROVED** [review]

**Pull request merged**